pkg/controller/certificates/certificate_controller.go

```diff
@@ -181,15 +173,14 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	defer func() {
 		glog.V(4).Infof("Finished syncing certificate request %q (%v)", key, time.Now().Sub(startTime))
 	}()
-	obj, exists, err := cc.csrStore.Store.GetByKey(key)
+	csr, err := cc.csrLister.Get(key)
```

Comment:
The change in this controller isn't safe.  Calls like `AutoApprove` are mutating.  You'll need to copy in some cases.